shared/utils.js

```
@@ -0,0 +1,204 @@
+            }
+        },
+
+        async queryGraphQl(query, variables) {
+            const reqOptions = this.buildRequestOptions(
+                `/${config.ctp.projectKey}/graphql`,
+                'POST',
+                {query, variables}
+            )
+
+            return ctpClient.execute(reqOptions)
+        },
+
+        buildUri(variables, endpoint) {
+            const uri = this.builder[endpoint]
+            if (variables.where && variables.where.length > 0)
+                uri.where(variables.where)
+            uri.perPage(variables.limit)
+            if (variables.sort)
+                uri.sort(variables.sort, true)
+            if (variables.expand)
+                uri.expand(variables.expand)
+            uri.withTotal(false)
+
+            return uri
+        },
+
+        fetch(uri) {
+            return ctpClient.execute(this.buildRequestOptions(uri.build()))
+        },
+
+        update(uri, id, version, actions) {
+            const body = {
+                version,
+                actions
+            }
+            return ctpClient.execute(
+                this.buildRequestOptions(uri.byId(id).build(), 'POST', body)
+            )
+        },
+
+        delete(uri, id, version) {
+            return ctpClient.execute(this.buildRequestOptions(
+                uri.byId(id).withVersion(version).build(),
+                'DELETE'
+            ))
+        }
+    }
+    return _.merge(customMethods, ctpClient)
+}
+
+export function initConfigOptions() {
+    const configOptions = {
+        ctp : {
+            projectKey : process.env.PROJECT_KEY,
+            clientId : process.env.CLIENT_ID,
+            clientSecret : process.env.CLIENT_SECRET
+        }
+    }
+
+    if (!configOptions.ctp.projectKey || !configOptions.ctp.clientId || !configOptions.ctp.clientSecret) {
+        console.error('Please set project key, client ID and client secret in environment variables')
+        return
```

I feel this might be better with throw instead of log, so you don't need to check if it's undefined or not in the script.

---

https://github.com/commercetools/commercetools-cli-scripts/pull/6/commits/44cbc6269804e0cfae63af8cf79f43b16f399eba